<launch>
  <arg name="plank_of_wood_id" default="2"/>
  <arg name="output" default="log"/>
  <arg name="state_estimation" default="false"/>

  <!-- Setup vision -->
  <group ns="vision">
    <param name="imageTopic"      value="/rgbd/rgb/high_res/image_rect" />
    <param name="cameraInfoTopic" value="/rgbd/rgb/high_res/camera_info" />
    <param name="parentNode"      value="talos/rgbd_rgb_optical_frame" />
    <param name="debugDisplay"    value="false" />
    <param name="broadcastTopic"  value="true" />
    <param name="broadcastTf"     value="true" />
    <param name="broadcastTfPostfix" value="_tf" />
    <param name="objectType" value="apriltag" />

    <node pkg="agimus_vision" type="tracker_object"
        output="$(arg output)"
        name="tag_tracker"
        />
    <!-- Table -->
    <node pkg="agimus_demos" type="group_of_tags.py" name="table_tags" args="
        --child-frame-format 'table/tag36_11_{:0>5d}'
        --measurement-parent-frame world
        --parent-frame table/base_link
        --no-description
        --size 0.060
        --tag 10
        --tag 11
        "
        output="$(arg output)"
        />

    <!-- Plank of wood 1 -->
    <node pkg="agimus_demos" type="group_of_tags.py" name="box_tags" args="
        --child-frame-format 'box/tag36_11_{:0>5d}'
        --measurement-parent-frame world
        --parent-frame box/base_link
        --no-description
        --size 0.060
        --tag $(arg plank_of_wood_id)0
        --tag $(arg plank_of_wood_id)1
        --tag $(arg plank_of_wood_id)2
        --tag $(arg plank_of_wood_id)3
        "
        output="$(arg output)"
        />
  </group>

  <group ns="agimus/hpp" if="$(arg state_estimation)">
    <param name="default_state_id"  value="2" />
    <!--<param name="use_configuration_constraint"  value="false" />-->
    <node pkg="agimus_hpp" type="estimation.py" name="hpp_estimation"
      args="--continuous-estimation --joint-state /joint_states" output="screen" />
  </group>
</launch>
